packages/local_auth/local_auth_android/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java

```diff
@@ -90,9 +92,27 @@ interface AuthCompletionHandler {
             .setConfirmationRequired((Boolean) call.argument("sensitiveTransaction"))
             .setConfirmationRequired((Boolean) call.argument("sensitiveTransaction"));
 
-    if (allowCredentials) {
-      promptBuilder.setDeviceCredentialAllowed(true);
-    } else {
+    boolean credentialsAllowed = false;
```

Comment:
Rather than duplicate this state, you can just check `allowedAuthenticators & DEVICE_CREDENTIAL`, so there's no chance of the bool and the actual authenticator list being out of sync.